project/Http4sSitePlugin.scala

```diff
@@ -0,0 +1,191 @@
+        ),
+        license = Some("Apache 2.0"),
+        documentationLinks = projectLinks,
+        projectLinks = Nil, // TODO
+        teasers = landingPage.teasers,
+      )
+
+    val teasers: Seq[Teaser] = Seq(
+      Teaser(
+        "Typeful",
+        "http4s servers and clients share an immutable model of requests and responses. Standard headers are modeled as semantic types, and entity codecs are done by typeclass.",
+      ),
+      Teaser(
+        "Functional",
+        "The pure functional side of Scala is favored to promote composability and easy reasoning about your code. I/O is managed through cats-effect.",
+      ),
+      Teaser(
+        "Streaming",
+        "http4s is built on FS2, a streaming library that provides for processing and emitting large payloads in constant space and implementing websockets.",
+      ),
+    )
+
+    val projectLinks: Seq[TextLink] = Seq(
+      TextLink.internal(Root / "versions.md", "Versions"),
+      TextLink.internal(Root / "changelog.md", "Changelog"),
+      TextLink.internal(Root / "getting-help.md", "Getting Help"),
+      TextLink.internal(Root / "contributing.md", "Contributing"),
+      TextLink.internal(Root / "adopters.md", "Adopters"),
+      TextLink.internal(Root / "code-of-conduct.md", "Code of Conduct"),
+      TextLink.internal(Root / "further-reading.md", "Further Reading"),
+    )
+
+  }
+
+  object versions {
+    def config: Versions = Versions(
+      currentVersion = current,
+      olderVersions = all.dropWhile(_ != current).drop(1),
+      newerVersions = all.takeWhile(_ != current),
+      renderUnversioned = true || current == v1_0,
+    )
+
+    private def version(version: String, label: String): Version =
+      Version(version, "v" + version, "/guide/quickstart.html", Some(label))
+
+    val v1_0: Version = version("1.0", "Dev")
+    val v0_23: Version = version("0.23", "Stable")
+    val v0_22: Version = version("0.22", "Stable")
+    val v0_21: Version = Version("0.21", "v0.21", "/index.html", Some("EOL"))
+    val choose: Version = Version(
+      "Help me choose...",
+      "",
+      "/versions.html",
+    ) // Pretend it's a "version" to get it into the menu
+
+    val all: Seq[Version] = Seq(v1_0, v0_23, v0_22, v0_21, choose)
+
+    val current: Version = v0_22
+  }
+
+  object redirects {
+
+    def theme = new ThemeProvider {
+      def build[F[_]: Sync]: Resource[F, Theme[F]] =
+        ThemeBuilder[F]("Http4s Redirects")
+          .addInputs( // add redirect htmls to the virtual file tree
+            redirects.foldLeft(InputTree[F]) { case (tree, (from, to)) =>
+              tree.addString(html(to), from / "index.html")
+            }
+          )
+          .build
+    }
+
+    def html(to: Path) =
+      s"""|<!DOCTYPE html>
+        |<meta charset="utf-8">
+        |<meta http-equiv="refresh" content="0; URL=$to">
+        |<link rel="canonical" href="$to">
+        |""".stripMargin
```

Comment:
There. That's not too bad, right?

Reply:
It's as good as GitHub is going to give us! 😆